packages/engine/src/simulation/package/state/packages/behavior_execution/chain.rs

```diff
@@ -65,7 +67,7 @@ impl IntoArrowChange for ChainList {
         let num_agents = range.end - range.start;
         let chains = &self.inner[range.start..range.end];
         let mut offsets = new_offsets_buffer(num_agents);
-        let mut_offsets = offsets.typed_data_mut::<i32>();
+        let mut_offsets = unsafe { offsets.typed_data_mut::<i32>() };
```

Comment:
Do we need to add a debug_assert or something to ensure safety on this? At least can we put a comment with `Safety` like we usually do